src/Publisher.ipp

```diff
@@ -0,0 +1,57 @@
+/**
+ * This software is distributed under the terms of the MIT License.
+ * Copyright (c) 2020 LXRobotics.
+ * Author: Alexander Entinger <alexander.entinger@lxrobotics.com>
+ * Contributors: https://github.com/107-systems/107-Arduino-Cyphal/graphs/contributors.
+ */
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include <array>
+
+/**************************************************************************************
+ * NAMESPACE
+ **************************************************************************************/
+
+namespace impl
+{
+
+/**************************************************************************************
+ * PUBLIC MEMBER FUNCTIONS
+ **************************************************************************************/
+
+template<typename T>
+bool Publisher::publish(T const & msg)
+{
+  CanardTransferMetadata const transfer_metadata =
+  {
+    .priority       = CanardPriorityNominal,
+    .transfer_kind  = CanardTransferKindMessage,
+    .port_id        = _port_id,
+    .remote_node_id = CANARD_NODE_ID_UNSET,
+    .transfer_id    = _transfer_id++,
+  };
+
+  /* Serialize message into payload buffer. */
+  std::array<uint8_t, T::MAX_PAYLOAD_SIZE> payload_buf;
+  payload_buf.fill(0);
+  size_t const payload_buf_size = msg.serialize(payload_buf.data());
```

Comment:
Got to check if this syntax is supported across all cores.

Reply:
`serialize` returns `0` in case of error (see [here](https://github.com/107-systems/107-Arduino-Cyphal/blob/fix-174/src/wrappers/uavcan/node/Heartbeat_1_0.hpp#L81)). I could return false in that case. Would that be sufficient for you?

Reply:
>serialize returns 0 in case of error

1. What if the message is zero-sized, like `uavcan.primitive.Empty`?
2. If serialization fails, you would be emitting a zero-length transfer, which may cause receivers to obtain invalid data due to the implicit zero extension rule.
3. In C++, there is usually no reason to define non-orthogonal-typed interfaces, as this is not C with its severe limitations on expressivity. Normally the serialize() method should return some algebraic type that encodes the possible outcomes: success with an `std::size_t` buffer size, or failure with an error code. With C++17 it would be `std::variant<std::size_t, enum class Error>`; in older versions you could use either some basic custom implementation or https://github.com/martinmoene/variant-lite.

Reply:
Hmm. I see your point. However, I'd file this as out-of-scope for this PR (this error in handling error messages isn't new) and attach it to https://github.com/107-systems/107-Arduino-Cyphal/issues/185 instead.